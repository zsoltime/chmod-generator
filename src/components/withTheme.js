import React from 'react';
import { ThemeProvider } from 'styled-components';

import defaultTheme from '../themes';

const withTheme = (
  WrappedComponent,
  theme = defaultTheme
) => props => (
  <ThemeProvider theme={theme}>
    <WrappedComponent {...props} />
  </ThemeProvider>
);

export default withTheme;
