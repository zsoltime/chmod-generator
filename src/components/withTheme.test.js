import React from 'react';
import { mount } from 'enzyme';

import withTheme from './withTheme';

describe('withTheme HOC', () => {
  it('wraps component in <ThemeProvider /> with default theme', () => {
    const Button = () => <button>Click Me</button>;
    const WrappedButton = withTheme(Button);
    const wrapper = mount(<WrappedButton />);

    expect(wrapper).toMatchSnapshot();
  });

  it('accepts a custom theme', () => {
    const Button = () => <button>Click Me</button>;
    const theme = { test: true };
    const WrappedButton = withTheme(Button, theme);
    const wrapper = mount(<WrappedButton />);

    expect(wrapper.find('ThemeProvider').prop('theme')).toMatchObject(
      theme
    );
  });

  it('passes props to wrapped component', () => {
    const Greeting = props => (
      <p>
        Hello, <span>{props.username}</span>!
      </p>
    );
    const username = 'john_doe';
    const WrappedGreeting = withTheme(Greeting);
    const wrapper = mount(<WrappedGreeting username={username} />);

    expect(wrapper.find('span').text()).toBe(username);
  });
});
