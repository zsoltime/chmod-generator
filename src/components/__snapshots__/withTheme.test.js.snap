// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withTheme HOC wraps component in <ThemeProvider /> with default theme 1`] = `
<Component>
  <ThemeProvider
    theme={
      Object {
        "background": "#f1f1ec",
        "black": "#363636",
        "blue": "#5c88f2",
        "border": "rgba(0, 0, 0, 0.1)",
        "checkbox": "rgba(255, 255, 255, .9)",
        "checkboxChecked": "#42a148",
        "focusRing": "#363636",
        "form": "rgba(255, 255, 255, .9)",
        "green": "#42a148",
        "grey": "#3a3a3a",
        "lightGrey": "rgba(0, 0, 0, 0.3)",
        "lighterGrey": "rgba(0, 0, 0, 0.1)",
        "link": "#42a148",
        "maxWidth": "600px",
        "offWhite": "#f1f1ec",
        "placeholder": "rgba(0, 0, 0, 0.3)",
        "text": "#363636",
        "transitionEase": "0.2s ease-in-out",
        "white": "rgba(255, 255, 255, .9)",
      }
    }
  >
    <Button>
      <button>
        Click Me
      </button>
    </Button>
  </ThemeProvider>
</Component>
`;
